qiskit/algorithms/amplitude_estimators/ae.py

```diff
@@ -297,15 +333,37 @@ def estimate(self, estimation_problem: EstimationProblem) -> "AmplitudeEstimatio
         result.num_evaluation_qubits = self._m
         result.post_processing = estimation_problem.post_processing
 
-        if self._quantum_instance.is_statevector:
+        measurements = OrderedDict()
+        samples = OrderedDict()
+        if self._sampler is not None:
+            circuit = self.construct_circuit(estimation_problem, measurement=True)
+            job = self._sampler.run([circuit])
+            if job.status() is not JobStatus.DONE:
+                raise AlgorithmError("The job was not completed successfully. ")
+            ret = job.result()
+            result.shots = 1
```

Comment:
This looks a little strange, is this for technical reasons later in the code or why is this set to 1? 🤔

With Hamamura-san's PR for default options in the primitives we can now extract the shots from the sampler like `sampler.run_options.get("shots")` 🙂